src/data/actions/assets.js

```diff
@@ -95,16 +105,20 @@ export const toggleLockAssetFailure = (asset, response) => ({
 export const toggleLockAsset = (asset, courseDetails) =>
   (dispatch) => {
     dispatch(togglingLockAsset(asset));
-    clientApi.requestToggleLockAsset(courseDetails.id, asset)
+    return clientApi.requestToggleLockAsset(courseDetails.id, asset)
       .then((response) => {
-        if (response.ok) {
-          dispatch(toggleLockAssetSuccess(asset));
-        } else {
-          dispatch(toggleLockAssetFailure(asset, response));
+        if (!response.ok) {
+          throw new Error(response);
         }
+      })
+      .then(() => {
+        dispatch(toggleLockAssetSuccess(asset));
+      })
+      .catch((error) => {
+        dispatch(toggleLockAssetFailure(asset, error));
       });
   };
 
-export const clearAssetsStatus = () =>
-  dispatch =>
-    dispatch({ type: assetActions.CLEAR_ASSETS_STATUS });
+export const clearAssetsStatus = () => ({
```

Comment:
I don't think this should be an async action creator, so I have changed it to be synchronous.